test/cases/compile_errors/overeager_int_comparison_elision.zig

```diff
@@ -0,0 +1,43 @@
+const std = @import("std");
+// Log to ensure the true or false branch of a conditional is analyzed.
+// Use a comptime src arg to ensure new analysis for each unique invocation.
+fn checkAnalyzed(comptime src: std.builtin.SourceLocation, comptime branch: bool, expr: bool) void {
```

Comment:
This doesn't seem like a good way to test this and also doesn't seem to have much value over the behavior test.

Reply:
It is a bit funny. But my motivation was testing for false positives. Should I just delete the test file, and we can expect false positives will be caught elsewhere?

Reply:
I don't think there is a possibility for false positives that wouldn't be caught by the behavior test since at least one branch of the if expression is evaluated and all of them have a `@compileError` in one branch.

Reply:
That... is a good point. I'll delete this and add your other suggestion now, thanks!